Feature/rspec test - Code Review

SantiagoGuerra opened the pull request:
Hello kind reviewer, in the following list you will find a general description of this project.

* We created test cases for each enumerable method
* Each method has positive and negative behavior
* We found some issues with the previous version and we fixed them to make them more similar like the original enumerator methods.
* We used `let` instead of before when creating variables

Thank you in advance to review this project.

euqueme committed:
RSPEC FILES

Aff rspec files and edit .rspect file to fetch the documentation when we execute the test cases

euqueme committed:
MY_EACH

Add the test_enumerables.rb file and wrote the first 2  test cases for the if and else cases of the my_each methdo in the main file

Also modify the my_each method so it returns self when block is given

disable rubocop for the spec_helper.rb file

the code has no rubocop ofenses

SantiagoGuerra committed:
MY_EACH_WITH_INDEX
add self

SantiagoGuerra committed:
MY_EACH_WITH_INDEX TEST
test method using two test cases
* When block is given
* When block is'nt given

euqueme committed:
MY_SELECT RSPEC

Created positive test cases when block is given and when block is not given

euqueme committed:
LET AND NEGATIVE TEST CASES

Switch to use let instead of before when creating variables.

Add negative test cases when an ArgumentError is raised for my_each, my_each_with_index and my_select

SantiagoGuerra committed:
MY_ALL?
change logic when a Regex is received

SantiagoGuerra committed:
MY_ALL? TEST
test my_all? enum method in different scenarios, all testcases works.

euqueme committed:
MY_ANY? AND MY_ALL?

Change validations in my_all and my_any methods so arguments are executed before blocks

create test cases for my_any from my_all but add two test cases more in both methods:
  * Validate that the arguments are validated when argument and block is given

* Validates when and argument that is not a class or a Regexp is given

euqueme committed:
MY_NONE? TEST

Create test cases for my_none? method from my_any? method but add one more validation to test that my_none?  is equal to the negation of my_any?

SantiagoGuerra committed:
MY_COUNT
improve conditional statment

SantiagoGuerra committed:
MY_COUNT TEST
test my_count enum method and add a new element to `array`.

all testcases passes.

euqueme committed:
MY_INJECT TEST

Change my_inject to take preference of argument over blocks

Create a test case for each of the conditions of this method and create test cases for negative cases

SantiagoGuerra committed:
MY_MAP TEST
test my_map? enum method and each testcase has been passed

whiz25 reviewed (approved):
## Awesome :clap: 
You guys have all tests passing so it's a :+1: from me. I wish you success in upcoming projects.
Check my comment and fix that request before merging.

### Martin Afani

whiz25 commented on spec/test_enumerables.rb:
```diff
@@ -0,0 +1,297 @@
+
+# frozen_string_literal: true
+
+require '../main.rb'
```
Try changing this to ```require_relative  '../main.rb'``` to fix the error when it's ```require '../main.rb'```

euqueme replied:
Solved!

Thank you very much.

euqueme committed:
REQUIRE_RELATIVE

Change require to require_relative so it works in both Linux and Windows computers.
